Shop.Web/Components/CategoriesName.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Shop.Data;
+using Shop.Web.Models.Category;
+using System.Linq;
+
+namespace Shop.Web.Components
+{
+    public class CategoriesName: ViewComponent
+    {
+        private readonly ICategory _categoryService;
+
+        public CategoriesName(ICategory categoryService)
+        {
+            _categoryService = categoryService;
+        }
+        public IViewComponentResult Invoke ()
+        {
+            var categories = _categoryService.GetAll().
+               Select(category => new CategoryDropdownModel
+               {
+                   Name = category.Name,
+                   Id = category.Id,
+               });
+
+            ViewBag.Categories = new CategoryDropdownIndexModel

```

Comment:
why did you put them into ViewBag? why can't you pass it into 

`var model = new CategoryIndex.....

return View(model)
`
?

Reply:
same here with naming, better would be CategoryMenuItemsModel